drivers/entropy/entropy_nrf5.c

```diff
@@ -278,11 +278,27 @@ static int entropy_nrf5_get_entropy_isr(struct device *dev, uint8_t *buf, uint16
 		nrf_rng_event_clear(NRF_RNG, NRF_RNG_EVENT_VALRDY);
 		nrf_rng_task_trigger(NRF_RNG, NRF_RNG_TASK_START);
 
+		/* Clear NVIC pending bit. This ensures that a subsequent
+		 * RNG event will set the Cortex-M single-bit event register
+		 * to 1 (the bit is set when NVIC pending IRQ status is
+		 * changed from 0 to 1)
+		 */
+		NVIC_ClearPendingIRQ(IRQN);
```

Comment:
So what happens if the ISR executes between these 2? It doesn't matter because we have the `nrf_rng_event_check()` later?

Reply:
@carlescufi the problem we try to solve here is the following: 
When the `get_entropy_isr()` function is called with the RNG interrupt masked, that is
- either IRQs are (globally)  locked, or
- the RNG IRQ is not enabled, or
- the function is called from a higher handler priority

then it is possible that the NVIC pending flag of the RNG interrupt is already set (to 1) before we do the first WFE. This means that the ARM core won't set the single-bit event register when the next RNG number is ready, since this is done when the NVIC pending IRQ flag switches from 0 to 1. This event register might be set due to other reasons, of course, but **if it is not, then the MCU cannot exit WFE when the next RNG number is ready**

> So what happens if the ISR executes between these 2?

Actually, the IRQ is disabled on line 274, so it cannot execute between the clear and the trigger. But it can be set to pending, which will set the event register and allow us to exit WFE. 

To summarize:
By adding an NVIC_ClearPendingIRQ(IRQN); statement before the first WFE, I guarantee that we will always witness a pending flag switch from 0 to 1, and this event will set the event-register bit to 1, allowing to exit WFE.

Reply:
@stephanosio maybe you could also take a quick look and review this reasoning?